### PR TITLE
[New eMIB Entry Process] Combination of new functionalities from my last PRs

### PR DESCRIPTION
**Pull request by fnormand01**
# Description

This PR introduced the following new functionalities combined  together:
- only one one page is displayed when you click on '_Star the eMIB Sample Test_' instead of having 5 pages of instruction.
- after clicking on '_Enter eMIB_' and then confirm in the popup box, you'll get to the instruction part of the test.
- the '**Background**' and '**Inbox**' tabs will be disabled.
- there will be only one button at bottom right in the test footer to start the test.
- once ready to start the test, after clicking on '_Start test_' and confirm in the popup box, the disabled tabs will become enabled.
- at the same time, the test footer will show the same two buttons we had before (Quit test and Submit test buttons).

## Type of change

- New feature (non-breaking change which adds functionality)

## Screenshot

**New eMIB Entry Process**
---
**Starting the _eMIB Sample Test_:**

image

**Enter eMIB:**

image

**Enter eMIB confirmation:**

image

**Show the instruction tab first + _Background_ and _Inbox_ tabs are disabled + Only _Start test_ button in the test footer:**

image

**Start test confirmation**

image

**Disabled tabs become enabled + _Quit test_ and _Submit test_ are now displayed in test footer**

image

# Testing

Manual steps to reproduce this functionality:

1.  Start the eMIB Sample Test.
2.  Follow the new entry process and make sure it looks as expected.
3.  Make also sure that it doesn't break anything.

# Checklist

Applicable for all code changes.

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new compiler warnings
- [ ] ~~My changes generate no new accessibility errors and/or warnings~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have researched WCAG2.1 accessibility standards and met them in this PR (can be navigated using a keyboard)
- [x] My changes look good on IE 11+ and Chrome

### CODE REVIEW

File: frontend/src/components/commons/TabNavigation.jsx [hunk 1]
```diff
@@ -29,12 +29,28 @@ const styles = {
   }
 };
 
+/* this function is used to disable specific tabs using an array of tab ids 
+this function is located here in order to be able to test it */
+export const isTabIdDisabled = (array, tabId) => {
```
Comment (caleybrock) [inline frontend/src/components/commons/TabNavigation.jsx]: A simpiler way is:
```
export const isTabIdDisabled = (array, tabId) => {
  // check if the tabId exists in the array
  return array.indexOf(tabId) > -1;
};
```

File: frontend/src/components/commons/TabNavigation.jsx [hunk 2]
```diff
@@ -46,14 +62,28 @@ class TabNavigation extends Component {
     this.setState({ currentTab: id, currentBody: this.props.tabSpecs[id].body });
   }
 
+  // this function is simply calling 'isTabIdDisabled' function with the needed parameters
+  handleDisabledTabCheck = tabId => {
```
Comment (caleybrock) [inline frontend/src/components/commons/TabNavigation.jsx]: You could just say:
```
  handleDisabledTabCheck = tabId => {
   return isTabIdDisabled(this.props.disabledTabsArray, tabId);
  };
```

File: frontend/src/components/commons/TabNavigation.jsx (hunk 1 above)
Comment (fnormand01) [reply frontend/src/components/commons/TabNavigation.jsx]: Fixed in commit #https://github.com/code-for-canada/project-thundercat/pull/251/commits/b25cbcc9d808533cac93562d871ee5b3f32e2848

File: frontend/src/components/commons/TabNavigation.jsx (hunk 2 above)
Comment (fnormand01) [reply frontend/src/components/commons/TabNavigation.jsx]: Fixed in commit #https://github.com/code-for-canada/project-thundercat/pull/251/commits/7201cc86b1d941ebcb235bb2df22119aa903a2b9

**Comment by fnormand01** [comment]
To answer your first comment. The switch between the _Instruction_ and _Background_ tabs will be done in another PR.

**Comment by caleybrock** [comment]
Looks good to merge!

**Pull request merged**